### PR TITLE
CI build step for integration tests

### PR DESCRIPTION
**Pull request by scholtzan**
When running integration tests it currently fails with: 

```
script/entrypoint: line 25: exec: pytest: not found
```

### CODE REVIEW

**Comment by scholtzan** [comment]
Is there a good way to test this locally?

**Comment by jklukas** [comment]
> Is there a good way to test this locally?

You can try using the CircleCI CLI: https://circleci.com/docs/2.0/local-cli/

It's been a while since I've used it. It was a pretty nice experience, but had some holes when creds, etc. are involved.

**Comment by scholtzan** [comment]
I guess that's also a good opportunity to tackle https://github.com/mozilla/bigquery-etl/issues/750

File: .circleci/config.yml [hunk 1]
```diff
@@ -57,6 +57,12 @@ jobs:
             echo "Cannot pass creds to forked PRs, so marking this step successful"
             circleci step halt
           fi
+    - &build
+      run:
+        name: Build
+        command: |
+          python3.8 -m venv venv/
+          venv/bin/pip install --upgrade -r requirements.txt
```
Comment (jklukas) [inline .circleci/config.yml:65]: The `&build` anchor is already defined earlier in this file, so we should be able to inject the contents here. I'm actually confused about the `<<: *build` line earlier in the `integration` job configuration, as that looks to be injecting content from the anchor, but at the wrong level here. I may have missed that in an earlier review. I can't remember exactly what the right yaml syntax should be here.

Comment (scholtzan) [reply .circleci/config.yml]: Figured it out.

**Comment by scholtzan** [comment]
I figured out how to run CircleCI successfully locally and added documentation.

**Pull request merged**